engine/src/rocksdb_engine.rs

```diff
@@ -0,0 +1,197 @@
+use std::{iter::repeat, path::PathBuf, sync::Arc};
+
+use rocksdb::{Options, WriteBatchWithTransaction, WriteOptions, DB};
+
+use crate::{
+    engine_api::{Delete, DeleteRange, Put, StorageEngine, WriteOperation},
+    error::EngineError,
+};
+
+/// `RocksDB` Storage Engine
+#[derive(Debug, Clone)]
+pub struct RocksEngine {
+    /// The inner storage engine of `RocksDB`
+    inner: Arc<DB>,
+}
+
+impl RocksEngine {
+    /// New `RocksEngine`
+    ///
+    /// # Errors
+    ///
+    /// Return `EngineError` when DB open failed.
+    #[inline]
+    pub fn new(data_dir: &PathBuf, tables: &[&'static str]) -> Result<Self, EngineError> {
```

Comment:
I think use `impl AsRef<Path>` is batter than `&PathBuf`, and maybe use `open` as this method's name is better, because we don't create new `DB` every time

Reply:
Fixed